src/main/kotlin/org/jetbrains/research/ml/coding/assistant/dataset/TaskTrackerDatasetFetcher.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.ml.coding.assistant.dataset
+
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import org.jetbrains.research.ml.coding.assistant.dataset.model.*
+import org.jetbrains.research.ml.coding.assistant.utils.getListFiles
+import java.io.File
+import kotlin.streams.toList
+
+object TaskTrackerDatasetFetcher : DatasetFetcher {
+    override fun fetchDataset(file: File): Dataset {
+        require(file.isDirectory) { "Argument has to be directory with tasks" }
+        val taskSolutions = file.getListFiles().parallelStream()
+            .filter { it.isDirectory }
+            .map(this::fetchTaskSolutions).toList()
+        return Dataset(taskSolutions)
+    }
+
+    override fun fetchTaskSolutions(file: File): TaskSolutions {
+        require(file.isDirectory) { "Argument has to be directory with solution files" }
+        val solutions = file.getListFiles().parallelStream()
+            .map(this::fetchDynamicSolution)
+            .filter { it.hasFinalSolution() }
+        return TaskSolutions(
+            datasetTask = DatasetTask.createFromString(file.name),
+            solutions?.toList() ?: listOf()
+        )
+    }
+
+    private fun fetchDynamicSolution(file: File): DynamicSolution {
```

Comment:
Please add some comment. I think it is not clear that is `DynamicSolution`. You can add some comment that the `DynamicSolution` exists in the Models. But I think you should describe why you need this function

Reply:
Honestly, I'm not sure what Nastya meant by this but okay